### PR TITLE
DO-3036: Touch up our 1GP GHA's template and documentation to ensure easy adoption

### PR DESCRIPTION
**Pull request by cmitzel-ncino**
- Removes redundant repo name from workflow name
- fixes inputs
- bumps action versions
- ran formatter
- cleans up commitlint stuff to also bring it more in line with other naming conventions
- changes `cancel-in-progress` to make it only cancel groups on non-release branches
- only print success message when actually succeeded

### CODE REVIEW

File: workflow-templates/1gp-ci.yml [hunk 1]
```diff
@@ -24,102 +24,101 @@ on:
...
       config-path:
-        required: true
+        required: false
         type: string
 
 concurrency:
-    group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
-    cancel-in-progress: true
+  group: "${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}"
+  cancel-in-progress: true
```
Comment (Kevmo92) [inline workflow-templates/1gp-ci.yml]:
```suggestion
  cancel-in-progress: ${{ github.ref != 'refs/heads/release' }}
```

File: workflow-templates/1gp-ci.yml [hunk 2]
```diff
@@ -24,102 +24,101 @@ on:
...
+          orgfarm-api-url: "${{REPLACE_ORGFARM_URL_SECRET}}"
+          preserve-org: "${{ inputs.PRESERVE_ORG }}"
+
+      # 🏁 Artifact Upload Phase 🏁
+      - name: 🤙 Upload Artifact and trigger packaging 🏄‍♂️
+        if: ${{ github.ref_name == 'release' }}
+        uses: ncino/action-aws/s3_trigger_pipeline@v2
+
+      - name: 👏 Success Message 👏
+        run: cat $DEVOPS_SCRIPTS_PATH/config/deploy_message.txt
```
Comment (Kevmo92) [inline workflow-templates/1gp-ci.yml]:
```suggestion
      - name: 👏 Success Message 👏
        if: ${{ success() }}
        run: cat $DEVOPS_SCRIPTS_PATH/config/deploy_message.txt
```

**Review by Kevmo92 — COMMENTED** [review]
Couple small suggestions

File: workflow-templates/1gp-ci.yml [hunk 3]
```diff
@@ -24,102 +24,102 @@ on:
...
-    ENABLE_LINTING: "false"
-    GIT_BASE_BRANCH: "${{ github.base_ref }}"
-    GIT_BRANCH: "${{ github.ref_name }}"
-    GITHUB_TOKEN: "${{REPLACE_TOKEN_SECRET}}"
-    REPO_NAME: "${{ github.event.repository.name }}"
+  BUILD_URL: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+  COMMIT_SHA1: "${{ github.sha }}"
+  DD_AGENT_MAJOR_VERSION: "7"
+  DD_API_KEY: "${{ REPLACE_DD_API_SECRET }}"
+  DD_APP_KEY: "${{ REPLACE_DD_APP_SECRET }}"
```
Comment (cheyenneralston-ncino) [inline workflow-templates/1gp-ci.yml:43]: Will these and other variables such as the `$GITHUB_TOKEN` not be the same between all projects? Should we just default them to something like `'${{ secrets.DD_API_KEY }}'`

File: workflow-templates/1gp-ci.yml [hunk 4]
```diff
@@ -24,102 +24,102 @@ on:
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      # 🎬 Install Phase 🎬
+      - name: 🗃️ Checkout Repo 🗃️
+        uses: actions/checkout@v4
+        with:
+          token: "${{ REPLACE_TOKEN_SECRET }}"
```
Comment (cheyenneralston-ncino) [inline workflow-templates/1gp-ci.yml:64]:
```suggestion
          token: "${{ env.GITHUB_TOKEN }}"
```

File: workflow-templates/1gp-ci.yml [hunk 5]
```diff
@@ -24,102 +24,102 @@ on:
...
+
+      - name: 👷‍♀️ Validate destructive changes 👷‍♀️
+        run: $DEVOPS_SCRIPTS_PATH/validate_destructive_changes.sh $COMMIT_SHA1
+
+      - name: 👩‍🌾 Get and Setup Org 👨‍🌾
+        id: setup-org
+        uses: ncino/action-org-setup/org-provision-action@v3
+        with:
+          capability: { { REPLACE_CAPABILITY_NAME } }
+          github-token: "${{ REPLACE_TOKEN_SECRET }}"
```
Comment (cheyenneralston-ncino) [inline workflow-templates/1gp-ci.yml:97]:
```suggestion
          github-token: "${{ env.GITHUB_TOKEN }}"
```

**Review by cheyenneralston-ncino — COMMENTED** [review]
Nice work! Few small comments/questions

**Comment by cmitzel-ncino** [comment]
@cheyenneralston-ncino I was just going off of what was already set up in the workflow, it was hiding the names of the secrets there. @Brian-Triplett is there like a potential security thing or were we doing because it felt right as a template?

**Comment by Cerelius** [comment]
> @cheyenneralston-ncino I was just going off of what was already set up in the workflow, it was hiding the names of the secrets there. @Brian-Triplett is there like a potential security thing or were we doing because it felt right as a template?

Security wanted us to not expose our secret names even though I feel like it's silly. I don't feel that it actually adds any real security.

**Review by Cerelius — APPROVED** [review]

**Review by ChaseBowen — APPROVED** [review]

**Pull request merged**